cmd/malwareExecuteScan_test.go

```diff
@@ -0,0 +1,169 @@
+			})
+		}
+	})
+
+	t.Run("Malware detected", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: "{\"malwareDetected\":true,\"encryptedContentDetected\":false,\"scanSize\":298782,\"mimeType\":\"application/octet-stream\",\"SHA256\":\"96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85\"}"}
+		error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+		assert.EqualError(t, error, "Malware scan failed for file 'target/myFile'. Malware detected: true, encrypted content detected: false")
+	})
+
+	t.Run("Encrypted content detected", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: "{\"malwareDetected\": false, \"encryptedContentDetected\": true, \"scanSize\":298782,\"mimeType\":\"application/octet-stream\",\"SHA256\":\"96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85\"}"}
+		error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+		assert.EqualError(t, error, "Malware scan failed for file 'target/myFile'. Malware detected: false, encrypted content detected: true")
+	})
+
+	t.Run("Malware and encrypted content detected", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: "{\"malwareDetected\": true, \"encryptedContentDetected\": true, \"scanSize\":298782,\"mimeType\":\"application/octet-stream\",\"SHA256\":\"96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85\"}"}
+		error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+		assert.EqualError(t, error, "Malware scan failed for file 'target/myFile'. Malware detected: true, encrypted content detected: true")
+	})
+}
+
+func TestFileNotFound(t *testing.T) {
+
+	httpClient := httpMock{StatusCode: 200}
+
+	open = func(path string) (io.ReadCloser, error) {
+		return nil, fmt.Errorf("open %s: no such file or directory", path)
+	}
+	defer func() { open = _open }()
+	error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+
+	assert.Error(t, error, "open target/myFile: no such file or directory")
+}
+
+func TestInternalServerError(t *testing.T) {
+
+	httpClient := httpMock{StatusCode: 500, ResponseBody: "internal server error"}
+
+	open = openFileMock()
+
+	defer func() { open = _open }()
+
+	err := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+
+	assert.Error(t, err, "Unexpected response code (500). 200 expected. Details: \"internal server error\"")
+}
+
+func TestRetrieveHash(t *testing.T) {
+
+	open = openFileMock()
+
+	defer func() {
+		open = _open
+	}()
+	hash, err := _getSHA256("target/myFile")
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, "3733cd977ff8eb18b987357e22ced99f46097f31ecb239e878ae63760e83e4d5", hash)
+	}
+}
+
+func openFileMock() func(path string) (io.ReadCloser, error) {
+	return func(path string) (io.ReadCloser, error) {
+		if path == "target/myFile" {
+			return readCloserMock{Content: "HELLO"}, nil // the content does not matter
+		}
+		return nil, fmt.Errorf("File '%s' not found", path)
```

Comment:
normally not. Anyway it is better to answer with an error compared to simply ignore that case. Maybe that is helpful later in case someone adust the code under test or the test here. I was not primarily to mimic the behaviour of a  open(), it is more about indicating a problem in the program flow. Nevertheless I adjusted ...